.github/workflows/build.yml

```diff
@@ -1,9 +1,14 @@
 name: Gradle Build
-on: [push, pull_request]
+on:
+  workflow_run:
```

Comment:
Trying to understand here. Shouldn't this workflow run on `push,pull`? Did we test that

Reply:
Probably need to give a bit more context here.  First, take a glance at issue [#4174](https://github.com/opensearch-project/OpenSearch/issues/4174) at the main repo.  I'm trying to replicate the same thing here.  My first comment on that issue references [these GitHub docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run).

Background: Jenkins gradle check is slow and takes a lot of resources.  It's not as slow as on main project (yet) but we are just getting started and once we add ITs it will probably get slower.  We are already pulling down the `feature/extensions` branch and publishing it on every run.  It's a waste to do that if the check is going to fail on spotless!

Goal:
 - Run one "fast" workflow that checks some of the obvious failures first (spotless, possibly add javadoc?)
 - Only if that check passes, run the full gradle check on Jenkins

Now to this specific change.  The `on` section indicates what triggers the workflow.  Previously it was either a `push` to main or a pull request.  The way that change is written, it will no longer run on every push or pull request it will only run on a `completed` `workflow_run` of the workflow named "Code Hygiene".   Later on it will also check the result of that workflow.

The Code Hygiene workflow is set to run on `push` and `pull_request` so it will trigger first, and when it completes, trigger this one.

It would be nice if you could see that workflow work on this PR, but for security reasons, the workflow_run only triggers on the default branch, so this will have to be merged for you to watch what happens.

We could get a little fancier and put `push` back on this workflow, and add an "or" conditional on line 11 below to run if it's a push workflow or if the workflow_run was a success.

Or we could be simple and always run gradle check and try to order the checks to fail fast (e.g., run linters BEFORE all the building activity, but I'm not sure they work without compiling so that still ends up taking a while.)

Or we could just always run the gradle check no matter how long it takes.

Reply:
Thanks for answering the question here. I see the design you are following here and I'm totally in line with the same. Just a small ask which you mentioned before already can we have `javadocs` with spotless because we have seen a couple of instance where gradle check fails because of `javadocs`. So quoting you again "fail fast". 😁

Reply:
Also, we should remove the linelint from OpenSearch repo too.